fastpair/rust/src/main.rs

```diff
@@ -16,18 +16,48 @@ use futures::executor;
 
 mod bluetooth;
 
-use bluetooth::{Adapter, Device};
+use bluetooth::{Adapter, Address, ClassicAddress, Device};
 
 fn main() -> Result<(), anyhow::Error> {
     let run = async {
         let mut adapter = bluetooth::default_adapter().await?;
         adapter.start_scan_devices()?;
 
-        while let Ok(device) = adapter.next_device().await {
-            println!("found {}", device.name()?)
-        }
+        while let Ok(ble_device) = adapter.next_device().await {
+            let name = ble_device.name()?;
+
+            if name.contains("LE_WF-1000XM3") {
+                println!("FOUND {} ", name);
+
+                let addr: Address = ble_device.address();
 
-        unreachable!("Done scanning");
+                // Dynamic dispatch is necessary here because `BleDevice` and
+                // `ClassicDevice` share the `Device` trait (and thus must have
+                // the same return type for `address()` method). This can be
+                // changed if `Device` trait should exclusively define
+                // cross-platform behavior.
+                let classic_addr = match addr {
+                    Address::Ble(ble) => ClassicAddress::try_from(ble),
+                    Address::Classic(_) => unreachable!(
```

Comment:
we should probably panic here instead of marking it unreachable, it'll be the same either way in terms of runtime but having unreachable in a potentially reachable condition seems wrong.